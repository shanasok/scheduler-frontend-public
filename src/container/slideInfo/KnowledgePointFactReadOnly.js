import React from 'react';
import TeacherEditPageContentIcon from "../teacher/TeacherEditPageContentIcon";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";

class KnowledgePointFactReadOnly extends React.Component {

    render() {
        const bulletPoint = '\u25CF';
        const fact = (this.props.content)? bulletPoint + ' ' + this.props.content.fact : "";
        const contentId = (this.props.content)? this.props.content.id : -1;
        const icons = (this.props.slideEditMode) ?
            <TeacherEditPageContentIcon pageContentId={contentId}/> : "";

        return <div className="knowledgePoint">
            {icons} <strong>{fact}</strong>
        </div>
    }
}

const mapStateToProps = state => ({
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({

        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(KnowledgePointFactReadOnly);