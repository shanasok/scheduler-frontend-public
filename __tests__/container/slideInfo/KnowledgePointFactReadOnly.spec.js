import React from 'react';
import KnowledgePointFactReadOnly from '../../../src/container/slideInfo/KnowledgePointFactReadOnly';
import renderer from 'react-test-renderer';
import {Provider as ReduxProvider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: []
}

const mockStore = configureMockStore()
const mockState = {
    lesson: lesson
}
let store = mockStore(mockState)
configure({adapter: new Adapter()});

it('renders correctly when fact is given', () => {

    const content = {
        contentType: "FACT",
        fact: "Test Fact",
        id: "1",
    }

    const tree = renderer
        .create(<ReduxProvider store={store}><KnowledgePointFactReadOnly content={content}/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[2].children).toContain('● ' + content.fact);
});

it('readonly renders correctly when fact is not given', () => {

    const tree = renderer
        .create(<ReduxProvider store={store}><KnowledgePointFactReadOnly/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[2].children).toContain("");
});

it('Shows icons when slideEditMode is true', () => {

    const mockStore1 = configureMockStore()
    const mockState1 = {
        lesson: lesson,
        slideEditMode: true
    }
    let store1 = mockStore1(mockState1)

    const tree = renderer
        .create(<ReduxProvider store={store1}><KnowledgePointFactReadOnly /></ReduxProvider>)
        .toJSON();
    expect(tree.children[0].type).toEqual('i');
});

it('Does not show icons when slideEditMode is false', () => {

    const mockStore1 = configureMockStore()
    const mockState1 = {
        lesson: lesson,
        slideEditMode: false
    }
    let store1 = mockStore1(mockState1)

    const tree = renderer
        .create(<ReduxProvider store={store1}><KnowledgePointFactReadOnly /></ReduxProvider>)
        .toJSON();
    expect(tree.children[0]).toEqual("");
});