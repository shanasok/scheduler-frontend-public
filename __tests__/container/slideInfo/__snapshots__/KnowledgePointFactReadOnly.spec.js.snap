// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readonly renders correctly when fact is not given 1`] = `
<div
  className="knowledgePoint"
>
  
   
  <strong>
    
  </strong>
</div>
`;

exports[`renders correctly when fact is given 1`] = `
<div
  className="knowledgePoint"
>
  
   
  <strong>
    ‚óè Test Fact
  </strong>
</div>
`;
